src/caps.rs

```diff
@@ -0,0 +1,48 @@
+use std::time::Duration;
+
+use anyhow::{Context, Result};
+use ed25519_dalek::{SigningKey, VerifyingKey};
+use iroh::NodeId;
+use rcan::{Capability, Expires, Rcan};
+use serde::{Deserialize, Serialize};
+use ssh_key::PrivateKey as SshPrivateKey;
+
+/// Potential capabilities for IPS
+#[derive(Ord, Eq, PartialOrd, PartialEq, Clone, Serialize, Deserialize, Debug)]
+#[repr(u8)]
+pub enum IpsCap {
+    /// API tokens, used in the RPC
+    Api,
+    /// Used to authenticate users.
+    Web,
+}
+
+impl Capability for IpsCap {
+    fn can_delegate(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Web, _) => false, // web can never delegate
+            (Self::Api, _) => false,
```

Comment:
This seems weird. I think this would disallow checking invocations with any kind of delegation altogether.
One of the checks in invocations is that if e.g. the `IpsCap::Web` is delegated, then invoked, that `Web.can_deleate(&Web)` is true.

I think `can_delegate` should *at least* be reflexive.

Reply:
> I think `can_delegate` should _at least_ be reflexive.

Why? the current scheme does not use delegations, so I wanted to encode that here

Reply:
Well, one problem is that this will prevent you from using the `Authorizer::check_invocation_from` function, since the capability you pass as an argument there will be checked against the capability in the *single* delegation you also pass along. And that'll be `false`, so `check_invocation_from` disallows that.